apps/nextjs-website/src/components/organisms/FragmentProvider/FragmentProvider.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type FragmentContextType = {
+  fragment: string | null;
+  setFragment: React.Dispatch<React.SetStateAction<string | null>>;
+};
+
+const FragmentContext = React.createContext<FragmentContextType | null>(null);
+
+type FragmentProviderProps = {
+  children: ReactNode;
+};
+
+export function FragmentProvider({ children }: FragmentProviderProps) {
+  const [fragment, setFragment] = useState<string | null>(() =>
+    typeof window !== 'undefined' ? window.location.hash : null
+  );
+
+  const fragmentChangeHandler = useCallback(() => {
+    setFragment(window.location.hash);
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener('hashchange', fragmentChangeHandler);
+    return () => {
+      window.removeEventListener('hashchange', fragmentChangeHandler);
+    };
+  }, []);
+
+  return (
+    <FragmentContext.Provider
+      value={{
+        fragment,
+        setFragment,
+      }}
+    >
+      {children}
+    </FragmentContext.Provider>
+  );
+}
+
+export function useFragment() {
+  const context = useContext(FragmentContext);
+  if (!context)
+    // eslint-disable-next-line functional/no-throw-statements
+    throw new Error(
```

Comment:
@datalek I did a test to verify that the provider throws a compile-time error if not used correctly and here is the result:
![Screenshot from 2023-08-29 09-54-22](https://github.com/pagopa/developer-portal/assets/39835990/c33fcae7-e3f2-4822-9e52-5f187940af87)